File: navigator/src/main/java/com/sha/kamel/navigator/model/ParcelableInfo.java
package com.sha.kamel.navigator.model;

import android.content.Intent;
import android.os.Parcelable;

public class ParcelableInfo {
    public Parcelable parcelable;
    public String name;

    public void addToIntent(Intent intent) {
        if (parcelable == null) return;
        intent.putExtra(name, parcelable);
    }
}

File: lib/src/main/java/com/sha/kamel/navigator/model/ParcelableInfo.kt
package com.sha.kamel.navigator.model

import android.content.Intent
import android.os.Parcelable

class ParcelableInfo {
    var parcelable: Parcelable? = null
    var name: String? = null

    fun addToIntent(intent: Intent) {
        if (parcelable == null) return
        intent.putExtra(name, parcelable)
    }
}
